distributed/tests/test_failed_workers.py

```diff
@@ -526,82 +527,6 @@ async def test_worker_time_to_live(c, s, a, b):
     set(s.workers) == {b.address}
 
 
-class SlowDeserialize:
-    def __init__(self, data, delay=0.1):
-        self.delay = delay
-        self.data = data
-
-    def __getstate__(self):
-        return self.delay
-
-    def __setstate__(self, state):
-        delay = state
-        import time
-
-        time.sleep(delay)
-        return SlowDeserialize(delay)
-
-    def __sizeof__(self) -> int:
-        # Ensure this is offloaded to avoid blocking loop
-        import dask
-        from dask.utils import parse_bytes
-
-        return parse_bytes(dask.config.get("distributed.comm.offload")) + 1
-
-
-@gen_cluster(client=True)
-async def test_handle_superfluous_data(c, s, a, b):
-    """
-    See https://github.com/dask/distributed/pull/4784#discussion_r649210094
-    """
-
-    def slow_deser(x, delay):
```

Comment:
This test has only recently been introduced in #4784 
It tried to provoke a race condition where a flight task runs into multiple transitions memory->release->compute but the test does not work reliably. Even on master it is kind of malfunctioning since the thread offloading does not work as intended (due to sizeof semantics).

I believe these edge cases are sufficiently tested in the `test_cancelled_state.py` module